src/compiler/checker.ts

```diff
@@ -35777,10 +35777,14 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         if (!result) {
             result = chooseOverload(candidates, assignableRelation, isSingleNonGenericCandidate, signatureHelpTrailingComma);
         }
+        const links = getNodeLinks(node);

```

Comment:
Maybe add a comment saying why we need this.

Reply:
done, I hope it helps - this case certainly is quite convoluted so the comment here was desperately needed

Reply:
thanks!